mrp_subcontracting_skip_no_negative/models/stock_move.py

```diff
@@ -35,6 +38,41 @@ def _action_done(self, cancel_backorder=False):
             ):
                 continue
             moves_with_no_check -= move
+            # If you have not been able to allocate previously it is because there is
+            # no stock, therefore it will leave the stock negative, we deduct the
+            # quantity checking the components and show the corresponding error.
+            test_condition = (
+                config["test_enable"] and self.env.context.get("test_stock_no_negative")
+            ) or not config["test_enable"]
+            if not test_condition:
+                continue
+            qty_precision = self.env["decimal.precision"].precision_get(
+                "Product Unit of Measure"
+            )
+            for p_move in unassigned_productions.move_raw_ids.filtered(
+                lambda x: x.state != "assigned"
+                and not x.product_id.allow_negative_stock
+                and not x.product_id.categ_id.allow_negative_stock
+                and not x.location_id.allow_negative_stock
+            ):
+                product = p_move.product_id.sudo()
+                location = p_move.location_id
+                location_qty = product.with_context(location=location.id).free_qty
+                new_qty = location_qty - p_move.product_uom_qty
+                if float_compare(new_qty, 0, precision_digits=qty_precision) == -1:
+                    raise ValidationError(
+                        _(
+                            "You cannot validate this stock operation because the "
+                            "stock level of the product '%s' would become negative "
+                            "(%s) on the stock location '%s' and negative stock is "
+                            "not allowed for this product and/or location."
```

Comment:
I think `stock level of the component product` is clearer than just `stock level of the product`, as this product isn’t in the move line. Specifying it as a component product helps avoid confusion. How do you think?